src/textual/_xterm_parser.py

```diff
@@ -54,36 +54,35 @@ def parse_mouse_code(self, code: str, sender: MessageTarget) -> events.Event | N
         if sgr_match:
             _buttons, _x, _y, state = sgr_match.groups()
             buttons = int(_buttons)
-            button = (buttons + 1) & 3
             x = int(_x) - 1
             y = int(_y) - 1
             delta_x = x - self.last_x
             delta_y = y - self.last_y
             self.last_x = x
             self.last_y = y
-            event: events.Event
-            if buttons & 64:
-                event = (
-                    events.MouseScrollUp if button == 1 else events.MouseScrollDown
-                )(sender, x, y)
-            else:
-                event = (
-                    events.MouseMove
-                    if buttons & 32
-                    else (events.MouseDown if state == "M" else events.MouseUp)
-                )(
-                    sender,
-                    x,
-                    y,
-                    delta_x,
-                    delta_y,
-                    button,
-                    bool(buttons & 4),
-                    bool(buttons & 8),
-                    bool(buttons & 16),
-                    screen_x=x,
-                    screen_y=y,
-                )
+            event_cls = (
+                (events.MouseScrollDown if buttons & 1 else events.MouseScrollUp)
+                if buttons & 64
+                else events.MouseMove
+                if buttons & 32
+                else (events.MouseDown if state == "M" else events.MouseUp)
+            )
+            if event_cls in (events.MouseScrollUp, events.MouseScrollDown):
```

Comment:
This condition is checking something that we only just assigned in the previous statement. Suggest if we need to do something different, we should to it in the branch where `event_cls` is assigned.